dl1_data_handler/reader.py

```diff
@@ -1,2028 +1,1206 @@
-from collections import Counter, OrderedDict
-import random
-import threading
+"""
+This module defines the ``DLDataReader`` and ``TableQualityQuery`` classes, which holds the basic reading and processing functionality for Deep Learning (DL) analyses.
+"""
+
+__all__ = [
+    "TableQualityQuery",
+    "DLDataReader",
+    "DLImageReader",
+    "get_unmapped_image",
+    "DLWaveformReader",
+    "get_unmapped_waveform",
+]
+
+from abc import abstractmethod
+from collections import OrderedDict
+from enum import Enum
 import numpy as np
-import pandas as pd
 import tables
+import threading
 
-from dl1_data_handler.image_mapper import ImageMapper
-from dl1_data_handler.processor import DL1DataProcessor
-
-import astropy.units as u
 from astropy.coordinates import SkyCoord
 from astropy.table import (
     Table,
-    join,  # let us merge tables horizontally
-    vstack,  # and vertically
+    unique,
+    join,
+    vstack,
 )
 
-from ctapipe.io import (
-    read_table,
-)  # let us read full tables inside the DL1 output file
-
-
-__all__ = ["DLDataReader"]
+from ctapipe.core import Component, QualityQuery
+from ctapipe.core.traits import (
+    Bool,
+    CInt,
+    Int,
+    IntTelescopeParameter,
+    Set,
+    List,
+    Path,
+    CaselessStrEnum,
+    Unicode,
+    TelescopeParameter,
+)
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.io import read_table
+from dl1_data_handler.image_mapper import ImageMapper
 
 
 lock = threading.Lock()
 
+class ProcessType(Enum):
+    Observation = "Observation"
+    Simulation = "Simulation"
+
+class TableQualityQuery(QualityQuery):
+    """Quality criteria for table-wise dl1b parameters."""
+
+    quality_criteria = List(
+        default_value=[
+            ("> 50 phe", "hillas_intensity > 50"),
+            ("Positive width", "hillas_width > 0"),
+            ("> 3 pixels", "morphology_n_pixels > 3"),
+        ],
+        allow_none=True,
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+
+class DLDataReader(Component):
+    """
+    Base component for reading and processing data from ctapipe HDF5 files for Deep Learning (DL) analyses.
+
+    This class handles the initialization and configuration of the data reader, including setting up quality criteria,
+    managing input files, and extracting relevant information from the data files. It supports both observational and
+    simulation data, and can operate in ``mono`` and ``stereo`` modes.
+
+    Attributes
+    ----------
+    quality_query : TableQualityQuery
+        An instance of TableQualityQuery to apply quality criteria to the data.
+    files : OrderedDict
```

Comment:
Why do you need an OrderedDict here? I don't see any use of specific features for it in the code. On the other hand, a standard dict since python 3.6 retains original order of items.